src/Microsoft.DotNet.Interactive.Journey/NotebookLessonParser.cs

```diff
@@ -64,29 +64,24 @@ public class NotebookLessonParser
             return CodeSubmission.Parse(content, "csharp", GetKernelNames(kernel));
         }
 
-        private static List<Microsoft.DotNet.Interactive.Documents.KernelName> GetKernelNames(CompositeKernel? kernel)
+        private static List<Documents.KernelName> GetKernelNames(CompositeKernel? kernel)
         {
-            List<Microsoft.DotNet.Interactive.Documents.KernelName> kernelNames = new();
+            List<Documents.KernelName> kernelNames = new();
 
             if (kernel is { })
             {
                 var kernelChoosers = kernel.Directives.OfType<ChooseKernelDirective>();
 
                 foreach (var kernelChooser in kernelChoosers)
                 {
-                    List<string> kernelAliases = new();
+                    var kernelAliases = kernelChooser.Aliases.Select(alias => alias[2..]).ToList();
 
-                    foreach (var alias in kernelChooser.Aliases)
-                    {
-                        kernelAliases.Add(alias[2..]);
-                    }
-
-                    kernelNames.Add(new Microsoft.DotNet.Interactive.Documents.KernelName(kernelChooser.Name[2..], kernelAliases));
+                    kernelNames.Add(new Documents.KernelName(kernelChooser.Name[2..], kernelAliases));
                 }
 
                 if (kernelNames.All(n => n.Name != "markdown"))
                 {
-                    kernelNames.Add(new("markdown", new[] { "md" }));
+                    kernelNames.Add(new Documents.KernelName("markdown", new[] { "md" }));
```

Comment:
Interesting that this "cleanup" adds qualifiers while others remove them. I think I'm getting soap in my eyes.

Reply:
We introduced 2 KernelName objects one is part of Interactive and the other one is part of document parsing the code was getting confused between the two types